### PR TITLE
Don't panic on indexed int64 fields

### PR DESCRIPTION
**Pull request by domodwyer**
See https://github.com/go-mgo/mgo/issues/475 for the mgo issue, and https://github.com/go-mgo/mgo/pull/476 for the PR.

Thanks to @mapete94 for committing the bugfix - a panic is always a nasty surprise!

### CODE REVIEW

**Comment by domodwyer** [comment]
I'll add a regression test later, hold off on the merge please :)

File: session.go [hunk 1]
```diff
@@ -1611,12 +1611,25 @@ func (idxs indexSlice) Swap(i, j int)      { idxs[i], idxs[j] = idxs[j], idxs[i]
 
 func simpleIndexKey(realKey bson.D) (key []string) {
 	for i := range realKey {
+		var vi int
 		field := realKey[i].Name
-		vi, ok := realKey[i].Value.(int)
-		if !ok {
+
```
Comment (szank) [inline session.go]: I haven't checked the call site, but is downcasting int64 to in  really necessary ? Mongo itself does handle 64bit integers IIRC.

Comment (domodwyer) [inline session.go]: For compatibility on 32 bit architectures I assume.

`int` is an `int64` on 64 bit systems.

File: README.md [hunk 2]
```diff
@@ -21,6 +21,7 @@ Further PR's (with tests) are welcome, but please maintain backwards compatibili
 * Improved multi-document transaction performance ([details](https://github.com/globalsign/mgo/pull/10), [more](https://github.com/globalsign/mgo/pull/11), [more](https://github.com/globalsign/mgo/pull/16))
 * Fixes cursor timeouts ([detials](https://jira.mongodb.org/browse/SERVER-24899))
 * Support index hints and timeouts for count queries ([details](https://github.com/globalsign/mgo/pull/17))
+* Don't panic when handling indexed `int64` fields ([detials](https://github.com/go-mgo/mgo/issues/475))
```
Comment (vgalu) [inline README.md]: s/detials/details/

File: session.go (hunk 1 above)
Comment (domodwyer) [reply session.go]: After looking at the code properly, the only valid values are "1" and "-1" so casting isn't an issue.

**Pull request merged**